### ISSUE TITLE
Refactor inferFooOp functions to use ArrayRef<int64_t> instead of DenseIntElementsAttr

### ISSUE THREAD

**Issue by burmako** [openxla/stablehlo#1031]
This will: 1) make things more consistent (we're already using `ArrayRef<int64_t>` for some parameters of inferFooOp functions), 2) make things easier to use (it's easier to whip up an ArrayRef than a DenseIntElementsAttr).

**Comment by burmako** [comment]
Assigning to Xin, since this falls into the overall theme of providing a public API for StableHLO.

**Comment by zhouxin913** [comment]
Agreed. And public API should follow similar pattern to provide easy to use interface like this.
